src/libs/actions/Report.js

```diff
@@ -404,6 +404,20 @@ function openReport(reportID, participantList = [], newReportObject = {}, parent
             // Add the createdReportActionID parameter to the API call
             params.createdReportActionID = optimisticCreatedAction.reportActionID;
         }
+
+        // If we are creating a thread, ensure the report action has childReportID property added
+        if (newReportObject.parentReportID && parentReportActionID) {
+            onyxData.optimisticData.push({
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${newReportObject.parentReportID}`,
+                value: {[parentReportActionID]: {childReportID: reportID}},
+            });
+            onyxData.failureData.push({
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${newReportObject.parentReportID}`,
+                value: {[parentReportActionID]: {childReportID: '0'}},
+            });
+        }
```

Comment:
This makes sure that when we create a report, the parent report action gets the childReportID key added. It should be updated by pusher, but if you are offline we would keep creating new threads by clicking on the preview or we could also crete new ones by quickly clicking on the preview before the pusher update comes through

Reply:
I ma not sure but I think no successData is needed here because we are only adding a property in ptimistic data and no pending value